### PR TITLE
Separate Reconcile functionality in a new function ReconcileNode

### PR DESCRIPTION
**Pull request by spectro30**
Signed-off-by: Md. Alif Biswas <alif@appscode.com>

### CODE REVIEW

**Comment by spectro30** [comment]
e2e test passed after the changes.

File: pkg/controller/secret.go [hunk 1]
```diff
@@ -57,7 +56,7 @@ func (c *Controller) ensureAuthSecret(db *api.MariaDB) error {
 	return c.upgradeAuthSecret(db)
 }
 
-func (c *Controller) createAuthSecret(db *api.MariaDB) (*core.LocalObjectReference, error) {
+func (c *Reconciler) createAuthSecret(db *api.MariaDB) (*core.LocalObjectReference, error) {
 	authSecretName := db.Name + "-auth"
```
Comment (kamolhasan) [inline pkg/controller/secret.go:60]: Use methods:

func (db *MariaDB) AuthSecretName() string {
 return NameWithSuffix(db.Name,"auth")
}

**Review by kamolhasan — CHANGES_REQUESTED** [review]
LGTM!

File: pkg/controller/secret.go (hunk 1 above)
Comment (spectro30) [reply pkg/controller/secret.go]: added

**Comment by spectro30** [comment]
General e2e-test passed.

**Review by kamolhasan — APPROVED** [review]
LGTM!

**Review by Emon46 — DISMISSED** [review]

**Review by kamolhasan — APPROVED** [review]
LGTM!

**Pull request merged**